### ISSUE TITLE
Crash when loading greasemonkey script without sufficient permissions

### ISSUE THREAD

**Issue by winny-** [qutebrowser/qutebrowser#7245]
**Version info**:
qutebrowser v2.5.1

**Does the bug happen if you start with `--temp-basedir`?**:
Yes, if one adds a greasemonkey script without sufficent permissions to the basedir's `greasemonkey` directory.

**Description**

**How to reproduce**

1. `touch ~/.local/share/qutebrowser/greasemonkey/crashme.js`
2. `chmod 0000 ~/.local/share/qutebrowser/greasemonkey/crashme.js`
3. Either restart Qutebrowser and observe the error popup OR run `:greasemonkey-reload` and observe a crash.

Screenshot of the error message upon startup:

![error-while-initializing](https://user-images.githubusercontent.com/3893828/173421552-407f3b17-98bd-4698-ad91-059f76e19954.png)

Trace from a crash using `:greasemonkey-reload`

```
==== Exception ====
Traceback (most recent call last):
  File "/nix/store/kf8gzqflmcvd3jzs9cvgy9y5hh5hspjv-qutebrowser-2.5.1/lib/python3.9/site-packages/qutebrowser/commands/runners.py", line 130, in run_safely
    self.run(text, count, safely=True)
  File "/nix/store/kf8gzqflmcvd3jzs9cvgy9y5hh5hspjv-qutebrowser-2.5.1/lib/python3.9/site-packages/qutebrowser/commands/runners.py", line 185, in run
    result.cmd.run(self._win_id, args, count=count)
  File "/nix/store/kf8gzqflmcvd3jzs9cvgy9y5hh5hspjv-qutebrowser-2.5.1/lib/python3.9/site-packages/qutebrowser/commands/command.py", line 551, in run
    self.handler(*posargs, **kwargs)
  File "/nix/store/kf8gzqflmcvd3jzs9cvgy9y5hh5hspjv-qutebrowser-2.5.1/lib/python3.9/site-packages/qutebrowser/browser/greasemonkey.py", line 462, in greasemonkey_reload
    scripts = gm_manager.load_scripts(force=force)
  File "/nix/store/kf8gzqflmcvd3jzs9cvgy9y5hh5hspjv-qutebrowser-2.5.1/lib/python3.9/site-packages/qutebrowser/browser/greasemonkey.py", line 313, in load_scripts
    with open(script_path, encoding='utf-8-sig') as script_file:
PermissionError: [Errno 13] Permission denied: '/tmp/qutebrowser-basedir-lunpsi15/data/greasemonkey/test.js'
```

**Comment by The-Compiler** [comment]
Pushed a fix. Will still need some tests, but I have to run now, so I'll keep the issue open for now.

**Comment by The-Compiler** [comment]
Decided to push a hacky fix for v2.5.2 anyways, and it's also fixed properly in master.
